### PR TITLE
RestAPIs for remove, list and detail instances

### PR DESCRIPTION
**Pull request by tuanlc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +77,19 @@ curl --header 'Host: http://linagora.beta.data.gouv.fr' http://127.0.0.1
...
+curl -XDELETE '127.0.0.1:8080/api/deployments/:requestId'
+```
+
+List all instances
+
+```
+curl -XGET '127.0.0.1:8080/api/deployments'
+```
+
+Get instace's detail by send a GET request:
```
Comment (sngo25) [inline README.md]: typo

File: backend/webserver/api/deployment/controller.js [hunk 2]
```diff
@@ -53,21 +53,29 @@ function create(req, res) {
...
 function remove(req, res) {
-  if (!req.query.domain) {
-    return res.status(400).json({ code: 400, message: 'Bad request', details: 'Missing domain' });
-  }
+  const requestId = req.params.requestId;
 
-  nginx.remove(req.query.domain, (err) => {
-    if (err) {
-      return res.status(500).json({ code: 500, message: 'Server error', details: err.message });
+  deploymentModule.findById(requestId).then((deployment) => {
```
Comment (sngo25) [inline backend/webserver/api/deployment/controller.js:58]: you need to add error handle

File: backend/webserver/api/deployment/controller.js [hunk 3]
```diff
@@ -53,21 +53,29 @@ function create(req, res) {
...
         return res.status(500).json({ code: 500, message: 'Server error', details: err.message });
       }
 
-      res.send('Removing Openpaas instance successfully');
+      swarm.remove(domain, err => {
+        if (err) {
+          return res.status(500).json({ code: 500, message: 'Server error', details: err.message });
+        }
+
+        res.send('Removing Openpaas instance successfully');
```
Comment (sngo25) [inline backend/webserver/api/deployment/controller.js]: `res.status(204).end();`

File: backend/webserver/api/deployment/index.js [hunk 4]
```diff
@@ -6,8 +6,12 @@ const router = express.Router();
...
 router.post('/', middleware.canCreate, middleware.hasValidRequestData, controller.create);
 
-router.post('/remove', controller.remove);
+router.delete('/:requestId', controller.remove);
 
 router.get('/:requestId/status', controller.getDeploymentStatus);
 
+router.get('/', controller.list);
+
+router.get(':requestId', controller.detail);
```
Comment (sngo25) [inline backend/webserver/api/deployment/index.js]: router.get(':requestId', controller.getDeployment);

File: backend/webserver/api/deployment/controller.js [hunk 5]
```diff
@@ -138,8 +146,40 @@ function getDeploymentStatus(req, res) {
...
+      return res.status(404).json({
+        status: 'not found'
+      });
+    }
+
+    return res.status(200).json(deployment);
+  });
+}
+
+function list() {
```
Comment (sngo25) [inline backend/webserver/api/deployment/controller.js]: no req, no res

File: backend/webserver/api/deployment/controller.js (hunk 2 above)
Comment (tuanlc) [reply backend/webserver/api/deployment/controller.js:58]: Done

File: backend/core/deployment.js [hunk 6]
```diff
@@ -14,8 +14,13 @@ function create(deployment) {
   return doc.save();
 }
 
+function list() {
+  return Deployment.find();
```
Comment (sngo25) [inline backend/core/deployment.js]:
```
function list(limit=20, offset=0) {
  return Deployment.find({})
    .limit(Number(limit))
    .skip(Number(offset));
```

**Pull request merged**